jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/completion/HandlerCompletionProvider.kt

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.completion
+
+import com.intellij.codeInsight.completion.CompletionParameters
+import com.intellij.codeInsight.completion.CompletionResultSet
+import com.intellij.codeInsight.lookup.CharFilter
+import com.intellij.openapi.project.Project
+import com.intellij.util.textCompletion.TextCompletionProvider
+import software.aws.toolkits.core.utils.error
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.core.utils.info
+import software.aws.toolkits.jetbrains.services.lambda.RuntimeGroup
+import software.aws.toolkits.jetbrains.services.lambda.runtimeGroup
+import java.lang.IllegalStateException
+
+class HandlerCompletionProvider(private val project: Project) : TextCompletionProvider {
+
+    private val logger = getLogger<HandlerCompletionProvider>()
+
+    // Suppress since gradle compiler mismatch return types for property delegate and fail build step
+    private val handlerCompletion: HandlerCompletion? by lazy {
+        val runtimeGroup = RuntimeGroup.determineRuntime(
+            project
+        )?.runtimeGroup
+        if (runtimeGroup == null) {
+            val message = "Unable to define Runtime Group for Lambda handler completion provider"
+            logger.error { message }
+            throw IllegalStateException(message)
+        }
+
+        return@lazy HandlerCompletion.getInstance(runtimeGroup) ?: let {
+            val message =
```

Comment:
probably doesn't need the assignment here

Reply:
Fixed.